tensorflow_addons/image/color_ops.py

```diff
@@ -79,51 +79,56 @@ def equalize(image: TensorLike, name: Optional[str] = None) -> tf.Tensor:
         return from_4D_image(image, image_dims)
 
 
-def sharpness_image(image: TensorLike, factor: Number) -> tf.Tensor:
+def _sharpness_image(image: TensorLike, factor: Number) -> tf.Tensor:
     """Implements Sharpness function from PIL using TF ops."""
     orig_image = image
     image_dtype = image.dtype
-    # SMOOTH PIL Kernel.
+    image_channels = image.shape[-1]
     image = tf.cast(image, tf.float32)
+
+    # SMOOTH PIL Kernel.
     kernel = (
         tf.constant(
             [[1, 1, 1], [1, 5, 1], [1, 1, 1]], dtype=tf.float32, shape=[3, 3, 1, 1]
         )
         / 13.0
     )
-    # Tile across channel dimension.
-    kernel = tf.tile(kernel, [1, 1, 3, 1])
-    strides = [1, 1, 1, 1]
+    kernel = tf.tile(kernel, [1, 1, image_channels, 1])
+
+    # Apply kernel channel-wise.
     degenerate = tf.nn.depthwise_conv2d(
-        image, kernel, strides, padding="VALID", dilations=[1, 1]
+        image, kernel, strides=[1, 1, 1, 1], padding="VALID", dilations=[1, 1]
     )
-    degenerate = tf.clip_by_value(degenerate, 0.0, 255.0)
```

Comment:
I'd say we have to keep the clip here because we cann't know if users pass a valid image into the function.

Reply:
NVM. I think you're right.

Reply:
@WindQAQ Ah I see this is inconsistent with `blend`. My thought was if someone were to use an image scaled to `(-1, 1)` or `[-127.5, 127.5]`, this would cut off the bottom half and removing it would enable these scaled images as well. It should work anyways right now, because `blend` clips to `[0, 255]`, but it may make sense to remove the clip in blend to enable images in more scaling ranges. Gotta see whether this works with other ops that make use of `blend` though.

Reply:
Edit: Things still works with the "supported" image ranges, but zero-centered ranges will be cut off by `blend`.

Reply:
I think the clip should be removed because people might want to apply these ops during post-processing. In this case, those images are not supposed to be VALID IMO. Got to check if CV2/scipy have similar functionality.